### PR TITLE
chore: fix typos in spelling

### PR DESCRIPTION
**Pull request by jsoref**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.
---

### CODE REVIEW

File: client/src/pages/learn/data-visualization/data-visualization-projects/index.md [hunk 1]
```diff
@@ -5,6 +5,6 @@ superBlock: Data Visualization
 ---
 ## Introduction to the Data Visualization Projects
 
-These challenges let you test your data visualzation skills and how to transfer and use data using AJAX technologies.
+These challenges let you test your data visualization skills and how to transfer and use data using AJAX technologies.
```
Comment (jsoref) [inline client/src/pages/learn/data-visualization/data-visualization-projects/index.md]: Most of this corpus appears to be en-GB. This file is en-US.

File: client/utils/formatting.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ const preFormatted = {
   ignorecase: 'ignoreCase',
   io: 'IO',
   isarray: 'isArray',
-  isealed: 'isSealed',
+  issealed: 'isSealed',
```
Comment (jsoref) [inline client/utils/formatting.js]: I'm not sure what this is doing. Potentially this is an API change.

File: curriculum/CHANGELOG.md [hunk 3]
```diff
@@ -22,7 +22,7 @@
 * **challenges:** adding code tags to description ([57d5b55](https://github.com/freeCodeCamp/curriculum/commit/57d5b55)), closes [#17911](https://github.com/freeCodeCamp/curriculum/issues/17911)
 * **challenges:** adding negative integer to challenge to improve tests ([#211](https://github.com/freeCodeCamp/curriculum/issues/211)) ([2adc516](https://github.com/freeCodeCamp/curriculum/commit/2adc516))
 * **challenges:** allow user to comment out undesired code ([72c2407](https://github.com/freeCodeCamp/curriculum/commit/72c2407))
-* **challenges:** challenge description is formatted and concised ([dcd8e45](https://github.com/freeCodeCamp/curriculum/commit/dcd8e45))
+* **challenges:** challenge description is formatted and concise ([dcd8e45](https://github.com/freeCodeCamp/curriculum/commit/dcd8e45))
```
Comment (jsoref) [inline curriculum/CHANGELOG.md]: I have no idea what this means.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/add-items-using-splice.english.md [hunk 4]
```diff
@@ -33,8 +33,8 @@ We have defined a function, <code>htmlColorNames</code>, which takes an array of
 
 ```yml
 tests:
-  - text: <code>htmlColorNames</code> should return <code>["DarkSalmon", "BlanchedAlmond", "LavenderBlush", "PaleTurqoise", "FireBrick"]</code>
-    testString: assert.deepEqual(htmlColorNames(['DarkGoldenRod', 'WhiteSmoke', 'LavenderBlush', 'PaleTurqoise', 'FireBrick']), ['DarkSalmon', 'BlanchedAlmond', 'LavenderBlush', 'PaleTurqoise', 'FireBrick']);
```
Comment (jsoref) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/add-items-using-splice.english.md]: This family is odd. The color is misspelled in this list, and yet it claims to be html colors.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/everything-be-true.english.md [hunk 5]
```diff
@@ -31,10 +31,10 @@ tests:
     testString: 'assert.strictEqual(truthCheck([{"user": "Tinky-Winky", "sex": "male"}, {"user": "Dipsy"}, {"user": "Laa-Laa", "sex": "female"}, {"user": "Po", "sex": "female"}], "sex"), false);'
   - text: '<code>truthCheck([{"user": "Tinky-Winky", "sex": "male", "age": 0}, {"user": "Dipsy", "sex": "male", "age": 3}, {"user": "Laa-Laa", "sex": "female", "age": 5}, {"user": "Po", "sex": "female", "age": 4}], "age")</code> should return false.'
     testString: 'assert.strictEqual(truthCheck([{"user": "Tinky-Winky", "sex": "male", "age": 2}, {"user": "Dipsy", "sex": "male", "age": 0}, {"user": "Laa-Laa", "sex": "female", "age": 5}, {"user": "Po", "sex": "female", "age": 4}], "age"), false);'
-  - text: '<code>truthCheck([{"name": "Pete", "onBoat": true}, {"name": "Repeat", "onBoat": true}, {"name": "FastFoward", "onBoat": null}], "onBoat")</code> should return false'
```
Comment (jsoref) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/everything-be-true.english.md]: I hope changing this is ok... I'm not actively looking to see what things are doing...

File: curriculum/challenges/english/06-information-security-and-quality-assurance/quality-assurance-and-testing-with-chai/use-assert.isok-and-assert.isnotok.english.md [hunk 6]
```diff
@@ -29,7 +29,7 @@ tests:
   - text: All tests should pass.
     testString: getUserInput => $.get(getUserInput('url') + '/_api/get-tests?type=unit&n=2').then(data => {assert.equal(data.state,'passed'); }, xhr => { throw new Error(xhr.responseText); })
   - text: You should choose the right assertion - isOk vs. isNotOk.
-    testString: getUserInput => $.get(getUserInput('url') + '/_api/get-tests?type=unit&n=2').then(data => {  assert.equal(data.assertions[0].method, 'isNotOk', 'Null is falsey'); }, xhr => { throw new Error(xhr.responseText); })
```
Comment (jsoref) [inline curriculum/challenges/english/06-information-security-and-quality-assurance/quality-assurance-and-testing-with-chai/use-assert.isok-and-assert.isnotok.english.md]: Both spellings are used in the world. This spelling was only used once in this corpus.

File: curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-102-triangle-containment.english.md [hunk 7]
```diff
@@ -12,7 +12,7 @@ Consider the following two triangles:
 A(-340,495), B(-153,-910), C(835,-947)
 X(-175,41), Y(-421,-714), Z(574,-645)
 It can be verified that triangle ABC contains the origin, whereas triangle XYZ does not.
-Using triangles.txt (right click and 'Save Link/Target As...'), a 27K text file containing the co-ordinates of one thousand "random" triangles, find the number of triangles for which the interior contains the origin.
```
Comment (jsoref) [inline curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-102-triangle-containment.english.md]: This corpus used both spellings. The majority of the world and this corpus favor no `-`.

File: curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-278-linear-combinations-of-semiprimes.english.md [hunk 8]
```diff
@@ -7,7 +7,7 @@ forumTopicId: 301928
 
 ## Description
 <section id='description'>
-Given the values of integers 1 < a1 < a2 <... < an, consider the linear combinationq1a1 + q2a2 + ... + qnan = b, using only integer values qk ≥ 0.
+Given the values of integers 1 < a1 < a2 <... < an, consider the linear combination q1a1 + q2a2 + ... + qnan = b, using only integer values qk ≥ 0.
```
Comment (jsoref) [inline curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-278-linear-combinations-of-semiprimes.english.md]: I didn't see any reason to omit the (` `)

File: curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-35-circular-primes.english.md [hunk 9]
```diff
@@ -106,7 +106,7 @@ function circularPrimes(n) {
           continue;
         }
         else if (!primes[x]) {
-          // If the rotated value is 0 then its not a ciruclar prime, break the loop
+          // If the rotated value is 0 then it isn't a circular prime, break the loop
```
Comment (jsoref) [inline curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-35-circular-primes.english.md]: `its` and `it's` are not the same. As a general rule, it's a lot safer to write `isn't` when `is` is the verb, as you won't confuse them.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/count-the-coins.english.md [hunk 10]
```diff
@@ -37,7 +37,7 @@ Implement a function to determine how many ways there are to make change for a d
 tests:
   - text: <code>countCoins</code> should be a function.
     testString: assert(typeof countCoins === 'function');
-  - text: <code>countCoints()</code> should return 242.
```
Comment (jsoref) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/count-the-coins.english.md]: I try not to fix things which seem to be intentionally pointing to a spelling error, but this appears to just be a spelling error. (e.g. misspellings of `Roosevelt` and `function`)

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/sort-using-a-custom-comparator.english.md [hunk 11]
```diff
@@ -32,8 +32,8 @@ tests:
     testString: assert.deepEqual(lengthSorter(["Mine", "is", "going", "great"]), ["going", "great", "Mine", "is"], '<code>lengthSorter(["Mine", "is", "going", "great"])</code> should return <code>["going", "great", "Mine", "is"]</code>.');
   - text: <code>lengthSorter(["Have", "fun", "sorting", "!!"])</code> should return <code>["sorting", "Have", "fun", "!!"]</code>.
     testString: assert.deepEqual(lengthSorter(["Have", "fun", "sorting", "!!"]), ["sorting", "Have", "fun", "!!"], '<code>lengthSorter(["Have", "fun", "sorting", "!!"])</code> should return <code>["sorting", "Have", "fun", "!!"]</code>.');
-  - text: <code>lengthSorter(["Everthing", "is", "good", "!!"])</code> should return <code>["Everthing", "good", "!!", "is"]</code>.
```
Comment (jsoref) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/sort-using-a-custom-comparator.english.md]: I'm pretty sure that fixing this doesn't impact the code.

File: tools/challenge-md-parser/text-to-data.test.js [hunk 12]
```diff
@@ -6,7 +6,7 @@ const textToData = require('./text-to-data');
 describe('text-to-data', () => {
   const expectedField = 'description';
   const otherExpectedField = 'instructions';
-  const unexpectedField = 'does-not-exis';
```
Comment (jsoref) [inline tools/challenge-md-parser/text-to-data.test.js]: There doesn't appear to be a reason to omit the `t`

File: curriculum/CHANGELOG.md (hunk 3 above)
Comment (RandellDawson) [reply curriculum/CHANGELOG.md]: I do not think you should be be making the CHANGELOG.md file

**Comment by raisedadead** [comment]
I am blocking this for a merge because, this involves changes to the code including functions and api endpoints. 

We need to meticulously QA the changes, especially if the endpoints are public?

**Comment by ojeytonwilliams** [comment]
I had a quick glance at this, too.  While nothing looked wrong, it would be easier to review if this was split into separate PRs.  All the minor changes (comments, typos in challenges and so on) could in principle go in one PR and the rest in another PR (or several).  Just so we know how carefully to QA something.

**Comment by raisedadead** [comment]
Well the good news is the OP has made individual commits for each typo change, so in theory we should be able to cherry-pick commits as we need them. But yeah, separate PRs would have been nice.

**Comment by jsoref** [comment]
I can split as https://github.com/freeCodeCamp/freeCodeCamp/pull/38100#issuecomment-581835680 described, but it's pretty much a one-off, doing that wipes out the individual commits noted in https://github.com/freeCodeCamp/freeCodeCamp/pull/38100#issuecomment-581840961.

If people want me to drop things, e.g. https://github.com/freeCodeCamp/freeCodeCamp/pull/38100#discussion_r369917467 please let me know. 

Anyway, when you reach a consensus about what you want me to do, let me know.

**Review by raisedadead — APPROVED** [review]
I have reviewed over 30 files related to platform, api, etc. Can someone go through the curricular content.

**Comment by RandellDawson** [comment]
@raisedadead I have also reviewed all the changes in this PR and everything looks fine.

There are 3 challenges (listed below) that will need to have the non-English versions `text` in the tests updated, since code references were changed.  I will create a separate PR to handle those now and reference this PR.  The reason these need to be updated also, is to keep things in sync for the changes being made in PR #38040

* [Add Items Using splice()](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/add-items-using-splice.english.md)

* [Everything Be True](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/everything-be-true.english.md)

* [Count the coins](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/08-coding-interview-prep/rosetta-code/count-the-coins.english.md)

**Comment by raisedadead** [comment]
Perfect. Thanks for the QA. I will have this merged shortly.

**Comment by raisedadead** [comment]
#### :sparkles: :sparkles: **AWESOME!** :sparkles: :sparkles: 

Thanks @jsoref, for this pull request and contributing to the code-base for the first time.
We will be looking forward for more contributions in future.

Cheers and happy contributing!

**Pull request merged**